Mathlib/Probability/Moments/SubGaussian.lean

```diff
@@ -0,0 +1,386 @@
+
+### Conditionally sub-Gaussian random variables and kernels
+
+A related notion to sub-Gaussian random variables is that of conditionally sub-Gaussian random
+variables. A random variable `X` is conditionally sub-Gaussian in the sense of (v) with respect to
+a sigma-algebra `m` and a measure `μ` if for all `t : ℝ`, `exp (t * X)` is `μ`-integrable and
+the conditional mgf of `X` conditioned on `m` is almost surely bounded by `exp (c * t^2 / 2)`
+for some constant `c`.
+
+As in other parts of Mathlib's probability library (notably the independence and conditional
+independence definitions), we express both sub-Gaussian and conditionally sub-Gaussian properties
+as special cases of a notion of sub-Gaussianity with respect to a kernel and a measure.
+
+## Main definitions
+
+* `Kernel.HasSubgaussianMGF`: a random variable `X` has a sub-Gaussian moment generating function
+  with parameter `c` with respect to a kernel `κ` and a measure `ν` if for `ν`-almost all `ω'`,
+  for all `t : ℝ`, the moment generating function of `X` with respect to `κ ω'` is bounded by
+  `exp (c * t ^ 2 / 2)`.
+* `HasCondSubgaussianMGF`: a random variable `X` has a conditionally sub-Gaussian moment generating
+  function with parameter `c` with respect to a sigma-algebra `m` and a measure `μ` if for all
+  `t : ℝ`, `exp (t * X)` is `μ`-integrable and the moment generating function of `X` contioned
+  on `m` is almost surely bounded by `exp (c * t ^ 2 / 2)` for all `t : ℝ`.
+  The actual definition uses `Kernel.HasSubgaussianMGF`: `HasCondSubgaussianMGF` is defined as
+  sub-Gaussian with respect to the conditional expectation kernel for `m` and the restriction of `μ`
+  to the sigma-algebra `m`.
+* `HasSubgaussianMGF`: a random variable `X` has a sub-Gaussian moment generating function
+  with parameter `c` with respect to a measure `μ` if for all `t : ℝ`, `exp (t * X)`
+  is `μ`-integrable and the moment generating function of `X` is bounded by `exp (c * t ^ 2 / 2)`
+  for all `t : ℝ`.
+  This is equivalent to `Kernel.HasSubgaussianMGF` with a constant kernel.
+  See `HasSubgaussianMGF_iff_kernel`.
+
+## Implementation notes
+
+### Definition of `Kernel.HasSubgaussianMGF`
+
+The definition of sub-Gaussian with respect to a kernel and a measure is the following:
+```
+structure Kernel.HasSubgaussianMGF (X : Ω → ℝ) (c : ℝ≥0)
+    (κ : Kernel Ω' Ω) (ν : Measure Ω' := by volume_tac) : Prop where
+  integrable_exp_mul : ∀ t, Integrable (fun ω ↦ exp (t * X ω)) (κ ∘ₘ ν)
+  mgf_le : ∀ᵐ ω' ∂ν, ∀ t, mgf X (κ ω') t ≤ exp (c * t ^ 2 / 2)
+```
+An interesting point is that the integrability condition is not integrability of `exp (t * X)`
+with respect to `κ ω'` for `ν`-almost all `ω'`, but integrability with respect to `κ ∘ₘ ν`.
+This is a stronger condition, as the weaker one did not allow to prove interesting results about
+the sum of two sub-Gaussian random variables.
+
+Fo the conditional case, that integrability condition reduces to integrability of `exp (t * X)`
```

Comment:
```suggestion
For the conditional case, that integrability condition reduces to integrability of `exp (t * X)`
```
conditional or unconditional here?

Reply:
I meant conditional, as written here, but that's actually true for both.